# Transcriptome wide analysis of classically and alternatively activated macrophages

## Summary

Next-generation sequencing (NGS) has revolutionized systems-based analysis of cellular pathways. The goals of this study are to compare NGS-derived human M1- and M2-like macrophage profiling (RNA-seq) to microarray and quantitative reverse transcription polymerase chain reaction (qRT–PCR) methods and to establish a high resolution transcriptome of human macrophages.

Total RNA was isolated from classically and aternative activated human macrophages.mRNA profiles were generated by deep sequencing og M1 and M2 macrophages from 3 donors using Illumina HiSeqSQ. The sequence reads that passed quality filters were analyzed at the transcript isoform level with two methods: Casava1.8 and TopHat followed by Cufflinks. qRT–PCR validation was performed using LightCycler and SYBR Green assays.

|     |title |geo_accession |source_name_ch1                    |organism_ch1 |Run       |SRA_Sample |cell_type                                |group |
|:----|:-----|:-------------|:----------------------------------|:------------|:---------|:----------|:----------------------------------------|:-----|
|M1_1 |M1_1  |GSM907013     |human monocyte derived macrophages |Homo sapiens |SRR452328 |SRS306689  |M1 macrophages (classically activated)   |M1    |
|M1_2 |M1_2  |GSM907014     |human monocyte derived macrophages |Homo sapiens |SRR452329 |SRS306690  |M1 macrophages (classically activated)   |M1    |
|M1_3 |M1_3  |GSM907015     |human monocyte derived macrophages |Homo sapiens |SRR452330 |SRS306691  |M1 macrophages (classically activated)   |M1    |
|M2_1 |M2_1  |GSM907016     |human monocyte derived macrophages |Homo sapiens |SRR452331 |SRS306692  |M2 macrophages (alternatively activated) |M2    |
|M2_2 |M2_2  |GSM907017     |human monocyte derived macrophages |Homo sapiens |SRR452332 |SRS306693  |M2 macrophages (alternatively activated) |M2    |
|M2_3 |M2_3  |GSM907018     |human monocyte derived macrophages |Homo sapiens |SRR452333 |SRS306694  |M2 macrophages (alternatively activated) |M2    |

## Goal

The goal of this analyis is to perform differential gene analysis between human M1- and M2-like macrophages. This will allow to identify M1 and M2 gene signatures for GSEA.

## Analysis and reports

1. Raw fastq files have been downloaded from [GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE36952) by running bash [script](scripts/bin/fastdump.sh). Meta data has been retrieved by running R [script](/scripts/bin/importGEOdata.R). Raw files were stored in /home/michal/data/GEO/GSE36952.
2. RNA-seq [nf-core](https://github.com/nf-core/rnaseq) data analysis pipeline incuding QC analysis, trimming, mapping to the genome (GRCh37) and generating gene count table was run using following bash [script](scripts/bin/nextflow.sh). Nextlow output was saved to /home/michal/data/nextflow/GSE3695.

* [nextflow workflow report](https://zhe-lab-ihit.github.io/GSE36952_RNAseq/scripts/data/execution_report.html)
* [results description](https://zhe-lab-ihit.github.io/GSE36952_RNAseq/scripts/data/results_description.html)
* [multiqc report](https://zhe-lab-ihit.github.io/GSE36952_RNAseq/scripts/data/GSE36952_v6_multiqc_report.html)

3. Following [R script](x) was used to retrieve gene counts and generate summarized experiment object.
4. The report containing differential gene analysis was prepared in [R markdown](x). M1/M2 gene sets were saved to [csv file]().

